.github/workflows/build.yml

```diff
@@ -0,0 +1,43 @@
+# Automatically build the project and run any configured tests for every push
+# and submitted pull request. This can help catch issues that only occur on
+# certain platforms or Java versions, and provides a first line of defence
+# against bad commits.
+
+name: build
+on: [pull_request, push]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        # Use these Java versions
+        java: [
+          21,    # Is a Current Java LTS & 17 is minimum supported by Minecraft
+        ]
+        # and run on both Linux and Windows
+        os: [windows-2022]
+        # to run on Linux: os: [ubuntu-22.04] (if want artifacts from Linux, other changes needed)
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: checkout repository
+        uses: actions/checkout@v4
+      - name: validate gradle wrapper
+        uses: gradle/actions/wrapper-validation@v4
+      - name: setup jdk ${{ matrix.java }}
+        uses: actions/setup-java@v4
+        with:
+          java-version: ${{ matrix.java }}
+          distribution: 'oracle'
+          # by default Github uses Microsoft one to restore change to 'microsoft'
+#      - name: make gradle wrapper executable
+#        if: ${{ runner.os != 'Windows' }}
+#        run: chmod +x ./gradlew
+# not needed on Windows
```

Comment:
Could we remove comments from the YAML that aren't relevant to this workflow please? This one and the ones mentioning Linux and Java versions above?

Also, I'm unfamiliar. Is `matrix` just "constants" for the sake of reference and reusability? Is this convention for workflows?